test/orange/orangeinp/ConvexRegion.test.cc

```diff
@@ -371,6 +371,145 @@ TEST_F(EllipsoidTest, standard)
     EXPECT_VEC_SOFT_EQ((Real3{3, 2, 1}), result.exterior.upper());
 }
 
+//---------------------------------------------------------------------------//
+// GENTRAP
+//---------------------------------------------------------------------------//
+using GenTrapTest = ConvexRegionTest;
+
+TEST_F(GenTrapTest, construct)
+{
+    // validate contruction parameters
+    EXPECT_THROW(GenTrap(-3,
+                         {{-1, -1}, {-1, 1}, {1, 1}, {1, -1}},
+                         {{-2, -2}, {-2, 2}, {2, 2}, {2, -2}}),
+                 RuntimeError);  // negative dZ
+    EXPECT_THROW(GenTrap(3,
+                         {{-1, -1}, {-1, 1}, {1, 1}, {2, 0}, {1, -1}},
+                         {{-2, -2}, {-2, 2}, {2, 2}, {2, -2}}),
+                 RuntimeError);  // 5 pts in -dZ
+    EXPECT_THROW(GenTrap(3,
+                         {{-1, -1}, {0.4, -0.4}, {1, 1}, {1, -1}},
+                         {{-2, -2}, {-2, 2}, {2, 2}, {2, -2}}),
+                 RuntimeError);  // non-convex
+
+    // a general, non-planar GenTrap, with 'twisted' faces - not yet
+    // implemented
+    EXPECT_THROW(GenTrap(3,
+                         {{-10, -10}, {-10, 10}, {10, 10}, {9, -11}},
+                         {{-10, -10}, {-10, 10}, {10, 10}, {10, -10}}),
+                 DebugError);
+}
+
+TEST_F(GenTrapTest, box_like)
+{
+    auto result = this->test(GenTrap(3,
+                                     {{-1, -1}, {-1, 1}, {1, 1}, {1, -1}},
+                                     {{-1, -1}, {-1, 1}, {1, 1}, {1, -1}}));
+
+    static char const expected_node[] = "all(+0, -1, -2, +3, +4, -5)";
+    static char const* const expected_surfaces[] = {"Plane: z=-3",
+                                                    "Plane: z=3",
+                                                    "Plane: x=-1",
+                                                    "Plane: y=1",
+                                                    "Plane: x=1",
+                                                    "Plane: y=-1"};
+
+    EXPECT_EQ(expected_node, result.node);
+    EXPECT_VEC_EQ(expected_surfaces, result.surfaces);
+    EXPECT_FALSE(result.interior) << result.interior;
+
+    // TODO: check why this is not lower=(-,-,-), upper=(+,+,+)
+    EXPECT_VEC_SOFT_EQ((Real3{1, 1, -3}), result.exterior.lower());
+    EXPECT_VEC_SOFT_EQ((Real3{-1, -1, 3}), result.exterior.upper());
```

Comment:
I think the orientation on your outward faces is flipped somehow: the `expected_node` shows that this is "below x=-1 and above x=1".

Reply:
Indeed, I was using counterclockwise points for both lower and upper z-faces.  
Reverted both to using clockwise.